src/lib/routes/dispatcher.js

```diff
@@ -1,18 +1,33 @@
-import permissionValidator from '../validators';
+import { permissionsValidator, requestValidator } from '../validators';
+import setRequestData from './setRequestData';
 
 const dispatcher = (controller, controllerMethod) => async (
   request,
   response,
   next
 ) => {
   try {
-    const { permissionRules } = controller;
+    const { permissionRules, validationRules } = controller;
+
+    // Get all request data and setup it in data param of request
+    setRequestData(request);
+
+    // Do request validation by schemas from validationRules
+    if (validationRules) {
+      await requestValidator(validationRules[controllerMethod.name], request);
```

Comment:
same here, validator should return data (with joi defaults), then we don't need a second set data request and you can rewrite it with a simple reassignment inside the condition body, avoiding unneeded setRequest if no validation rules were passed alltogether

Reply:
This function (dispatcher) will be called A LOT, like on every possible route in our app, so optimizing it is kinda important